AlfrescoApp/Model/Managers/RealmSyncManager+Refresh.m

```diff
@@ -0,0 +1,217 @@
+        nodeStatus.status = SyncStatusWaiting;
+    }
+}
+
+- (void)invalidateParentChildHierarchy
+{
+    RLMRealm *realm = self.mainThreadRealm;
+    RLMResults *allNodes = [[RealmManager sharedManager] allNodesInRealm:realm];
+    
+    for (RealmSyncNodeInfo *node in allNodes)
+    {
+        if (node.parentNode)
+        {
+            [realm beginWriteTransaction];
+            node.parentNode = nil;
+            [realm commitWriteTransaction];
+        }
+    }
+}
+
+- (void)rebuildDataBaseWithCompletionBlock:(void (^)(BOOL completed))completionBlock
+{
+    RLMRealm *realm = self.mainThreadRealm;
+    RLMResults *topLevelFolders = [[RealmManager sharedManager] topLevelFoldersInRealm:realm];
+    
+    self.nodeChildrenRequestsCount = 0;
+    
+    for (RealmSyncNodeInfo *node in topLevelFolders)
+    {
+        AlfrescoFolder *folder = (AlfrescoFolder *)node.alfrescoNode;
+        
+        [self retrieveNodeHierarchyForFolder:folder withCompletionBlock:^(BOOL completed){
+            if(self.nodeChildrenRequestsCount == 0)
+            {
+                if(completionBlock)
+                {
+                    completionBlock(completed);
+                }
+            }
+        }];
+    }
+}
+
+- (void)retrieveNodeHierarchyForFolder:(AlfrescoFolder *)folder withCompletionBlock:(void (^)(BOOL completed))completionBlock
+{
+    self.nodeChildrenRequestsCount++;
+    
+    [self.documentFolderService retrieveChildrenInFolder:folder completionBlock:^(NSArray *array, NSError *error){
+        self.nodeChildrenRequestsCount--;
+        
+        if (array)
+        {
+            for (AlfrescoNode *childNode in array)
+            {
+                [self updateDataBaseForChildNode:childNode withParent:folder];
+                
+                if (childNode.isFolder)
+                {
+                    [self retrieveNodeHierarchyForFolder:(AlfrescoFolder *)childNode withCompletionBlock:^(BOOL completed){
+                        if (completionBlock != NULL)
+                        {
+                            completionBlock(YES);
+                        }
+                    }];
+                }
+            }
+        }
+        else
+        {
+            RLMRealm *realm = self.mainThreadRealm;
+            RealmSyncNodeInfo *syncNodeInfo = [[RealmManager sharedManager] syncNodeInfoForObjectWithId:folder.identifier ifNotExistsCreateNew:NO inRealm:realm];
+            
+            [realm beginWriteTransaction];
+            syncNodeInfo.isTopLevelSyncNode = NO;
+            [realm commitWriteTransaction];
+        }
+        
+        if (completionBlock != NULL)
+        {
+            completionBlock(YES);
+        }
+    }];
+}
+
+- (void)updateDataBaseForChildNode:(AlfrescoNode *)childNode withParent:(AlfrescoNode *)parentNode
+{
+    NSString *childNodeIdentifier = [childNode syncIdentifier];
+    
+    RLMRealm *realm = self.mainThreadRealm;
```

Comment:
Same as the above comment - please get the realm instance from defaultRealm method instead of the property.